addons/sale_coupon/models/coupon_program.py

```diff
@@ -9,6 +9,27 @@ class CouponProgram(models.Model):
 
     order_count = fields.Integer(compute='_compute_order_count')
 
+    def apply_on_order(self, order, coupon_code):
+        self.ensure_one()
+
+        error_status = self._check_promo_code(order, coupon_code)
+        if not error_status:
+            if self.promo_applicability == 'on_next_order':
+                # Avoid creating the coupon if it already exist
+                if self.discount_line_product_id.id not in order.generated_coupon_ids.filtered(lambda coupon: coupon.state in ['new', 'reserved']).mapped('discount_line_product_id').ids:
```

Comment:
event if a move, split line to make it readable
```py
xx = order.generated_coupon_ids.filtered(lambda coupon: coupon.state in ['new', 'reserved']).mapped('discount_line_product_id').ids
if self..discount_line_product_id.id not in xx:
```

Reply:
Okay, but do you want it in a nested if so it only gets evaluated when necessary?